jekyll/_cci2/container-runner.adoc

```diff
@@ -95,6 +95,51 @@ agent:
           - name: "other"
 ```
 
+[#unsafe-retries]
+=== Unsafe retries
+
+Unsafe retries enable container runner to automatically rerun tasks if there are unexpected interruptions during their execution. These disruptions could be due to network connectivity issues, the underlying node shutting down, or other unpredictable causes. This feature can also be used when scheduling workloads on spot instances, which often come with cost-saving benefits at the risk of pod preemptions with many Kubernetes providers.
+
+[CAUTION]
+====
+This feature is called “unsafe retries” for a reason. Unlike automatic retries on startup, retrying tasks during runtime can be risky. This is because tasks can have arbitrary steps that produce side effects which are not idempotent or stateless. It is important to use unsafe retries knowing the impact it may have on jobs and workflows. Use this feature with jobs and workflows that are fault-tolerant.
+====
+
+Here is how unsafe retries work with container runner:
+
+1. If a pod fails or gets evicted during runtime, container runner will "un-claim", or release the task.
+2. All associated resources of the task are cleaned up.
+3. The released task then becomes available for reclaim by any container runner instance configured for the same resource class.
+4. Any previously ran steps will be reset, and the task will restart from scratch.
```

Comment:
The wording of "reset" here may also give the impression that we're undoing side effects somehow. Maybe simplify to "The task will be started from scratch, including previously ran steps".

I'm not sure about the usage of "task" vs "job" here. We don't really expose the concept of a task to customers

Reply:
We do expose the concept of tasks to customers: https://circleci.com/docs/faq/#what-is-a-CircleCI-task-vs-a-job

Also within this same document: https://circleci.com/docs/container-runner/#resource-class-configuration-custom-pod